cmd/clusterctl/clusterdeployer/bootstrap/kind/kind.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+
+	"k8s.io/klog"
+)
+
+const (
+	kubeconfigEnvVar = "KUBECONFIG"
+	kindClusterName  = "clusterapi"
+)
+
+type Kind struct {
+	options []string
+	// execFunc implemented as function variable for testing hooks
+	execFunc func(args ...string) (string, error)
+}
+
+func New() *Kind {
+	return WithOptions([]string{})
+}
+
+func WithOptions(options []string) *Kind {
+	return &Kind{
+		execFunc: execFunc,
+		options:  append(options, fmt.Sprintf("name=%s", kindClusterName)),
+	}
+}
+
+var execFunc = func(args ...string) (string, error) {
```

Comment:
this can be an actual function instead?
`func execFunc(...) ...`

Reply:
Replied below

Reply:
i guess my point was that this can still be regular function declaration.